matsim/src/main/java/org/matsim/core/events/SimStepParallelEventsManagerImpl.java

```diff
@@ -181,252 +157,115 @@ public void initProcessing() {
 	 */
 	@Override
 	public synchronized void finishProcessing() {
-		
-		/*
-		 * If an exception occurred, at least one of the events processing threads
-		 * has crashed. As a result, also all other events processing threads have
-		 * been stopped.
-		 * If not, it is waited until all threads have ended processing events.
-		 */
-		Throwable throwable = this.hadException.get();
-		if (throwable == null) {
-			try {
-				this.processEvent(new LastEventOfIteration(Double.POSITIVE_INFINITY));
-				iterationEndBarrier.await();
-			} catch (InterruptedException | BrokenBarrierException e) {
-				this.hadException.set(e);
-			}
-        }
-		
-		delegate.finishProcessing();
-		for (EventsManager eventsManager : this.eventsManagers) eventsManager.finishProcessing();
-		
-		/*
-		 * After the simulation Events are processed in
-		 * the Main Thread.
-		 */
-		this.parallelMode = false;
 
-		if (throwable != null) {
-			throw new RuntimeException("Exception while processing events. Cannot guarantee that all events have been fully processed.", throwable);
+		log.info("finishProcessing: Before awaiting all event processes");
```

Comment:
In the comment it says that processEvent(..) might be called after finishProcessing().
If this is really done from a different thread than I think a new event could be queued after the phaser advanced, but before init was set to false.

`isInitialized` is an atomic, but never any atomic operations like compareAndSwap are used, which is bit suspicious.